async-query-core/src/main/java/org/opensearch/sql/spark/validator/SQLQueryValidationVisitor.java

```diff
@@ -0,0 +1,609 @@
+    validateAllowed(GrammarElement.RESET);
+    return super.visitResetQuotedConfiguration(ctx);
+  }
+
+  @Override
+  public Void visitSetConfiguration(SetConfigurationContext ctx) {
+    validateAllowed(GrammarElement.SET);
+    return super.visitSetConfiguration(ctx);
+  }
+
+  @Override
+  public Void visitShowColumns(ShowColumnsContext ctx) {
+    validateAllowed(GrammarElement.SHOW_COLUMNS);
+    return super.visitShowColumns(ctx);
+  }
+
+  @Override
+  public Void visitShowCreateTable(ShowCreateTableContext ctx) {
+    validateAllowed(GrammarElement.SHOW_CREATE_TABLE);
+    return super.visitShowCreateTable(ctx);
+  }
+
+  @Override
+  public Void visitShowNamespaces(ShowNamespacesContext ctx) {
+    validateAllowed(GrammarElement.SHOW_NAMESPACES);
+    return super.visitShowNamespaces(ctx);
+  }
+
+  @Override
+  public Void visitShowFunctions(ShowFunctionsContext ctx) {
+    validateAllowed(GrammarElement.SHOW_FUNCTIONS);
+    return super.visitShowFunctions(ctx);
+  }
+
+  @Override
+  public Void visitShowPartitions(ShowPartitionsContext ctx) {
+    validateAllowed(GrammarElement.SHOW_PARTITIONS);
+    return super.visitShowPartitions(ctx);
+  }
+
+  @Override
+  public Void visitShowTableExtended(ShowTableExtendedContext ctx) {
+    validateAllowed(GrammarElement.SHOW_TABLE_EXTENDED);
+    return super.visitShowTableExtended(ctx);
+  }
+
+  @Override
+  public Void visitShowTables(ShowTablesContext ctx) {
+    validateAllowed(GrammarElement.SHOW_TABLES);
+    return super.visitShowTables(ctx);
+  }
+
+  @Override
+  public Void visitShowTblProperties(ShowTblPropertiesContext ctx) {
+    validateAllowed(GrammarElement.SHOW_TBLPROPERTIES);
+    return super.visitShowTblProperties(ctx);
+  }
+
+  @Override
+  public Void visitShowViews(ShowViewsContext ctx) {
+    validateAllowed(GrammarElement.SHOW_VIEWS);
+    return super.visitShowViews(ctx);
+  }
+
+  @Override
+  public Void visitUncacheTable(UncacheTableContext ctx) {
+    validateAllowed(GrammarElement.UNCACHE_TABLE);
+    return super.visitUncacheTable(ctx);
+  }
+
+  @Override
+  public Void visitFunctionName(FunctionNameContext ctx) {
+    validateFunctionAllowed(ctx.qualifiedName().getText());
+    return super.visitFunctionName(ctx);
+  }
+
+  private void validateFunctionAllowed(String function) {
+    FunctionType type = FunctionType.fromFunctionName(function.toLowerCase());
+    switch (type) {
+      case MAP:
+        validateAllowed(GrammarElement.MAP_FUNCTIONS);
+        break;
+      case CSV:
+        validateAllowed(GrammarElement.CSV_FUNCTIONS);
+        break;
+      case MISC:
+        validateAllowed(GrammarElement.MISC_FUNCTIONS);
+        break;
+      case UDF:
+        validateAllowed(GrammarElement.UDF);
+        break;
+    }
+  }
+
+  private void validateAllowed(GrammarElement element) {
+    if (!grammarElementValidator.isValid(element)) {
+      throw new IllegalArgumentException(element + " is not allowed.");
```

Comment:
add datasource info in exception? and add log.

Reply:
Added logging at upper level as this class doesn't know the datasourceType.